DummyProject/DummyProject/ViewController.swift

```diff
@@ -90,12 +146,69 @@ class ViewController: UIViewController {
         alertController.addAction(okAction)
         present(alertController, animated: true, completion: nil)
     }
+
+    @objc private func closeKeyboard() {
+        view.endEditing(true)
+    }
 }
 
+// MARK: - UITextField delegate
+
 extension ViewController: UITextFieldDelegate {
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        textField.endEditing(true)
+        closeKeyboard()
         return true
     }
 }
+
+// MARK: - UIPickerView dataSource & delegate
+
+extension ViewController: UIPickerViewDataSource, UIPickerViewDelegate {
+
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return WebPageType.allCases.count
+    }
+
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return WebPageType.allCases[row].rawValue
+    }
+
+    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        contentTextField.text = WebPageType.allCases[row].rawValue
+    }
+}
+
+// MARK: - WebPageViewController delegate
+
+extension ViewController: WebPageViewControllerDelegate {
+
+    func webViewCanGoBack() {
+        print("webViewCanGoBack")
```

Comment:
will these be removed in future? guessing they're just for debugging purposes

Reply:
I can remove all the loggings from the DummyProject. The reason for keeping them is to make it easier for QA verification because I'm not sure how QA test this.